### DISCUSSION TITLE
When I transfer files using rdpdr, the desktop is very laggy,Is there a solution?

### DISCUSSION THREAD

**Discussion by bestxpp** [FreeRDP/FreeRDP#10681] [Q&A] [answered]
I used the rdpdr plugin of FreeRDP to map a disk, and then I started copying files from the mapped disk to the remote host. During the copying process, the desktop became very laggy. I checked the bandwidth and found it was almost at 80-90 Mbps. I think the large file transfer is causing the desktop lag. I also tested with MSTSC, and it was still laggy. Is there any way to reduce the size of the file transfer when using rdpdr?

**Comment by bestxpp** [comment]
I debugged the RDPDR code and found that the file offset received in the IRP_MJ_READ event is always:
[file_id=1] length=1048576, offset=0
[file_id=1] length=1048576, offset=1048576
[file_id=1] length=1048576, offset=2097152

How can I modify the file and its length to reduce the bandwidth usage during file copying？

**Reply by akallabeth** [reply]
patch windows RDP server ;)

**Reply by bestxpp** [reply]
how to patch..  Are there no other ways?

**Reply by bestxpp** [reply]
I almost forgot, my server is running a Debian environment using xrdp, not a Windows server

**Reply by akallabeth** [reply, marked as answer]
@bestxpp then ask [xrdp](https://github.com/neutrinolabs/xrdp/) developers.
for `FreeRDP` the size of a single packet request is up to the implementing server calling `RdpdrServerContext::DriveReadFile` and `RdpdrServerContext::DriveWriteFile`
